include/unifex/connect_awaitable.hpp

```diff
@@ -92,12 +92,13 @@ class _sender_task<Receiver>::type {
       return awaiter{(Func &&) func};
     }
 
-    template(typename Value)
-      (requires callable<decltype(unifex::await_transform), promise_type&, Value>)
-    auto await_transform(Value&& value)
-        noexcept(is_nothrow_callable_v<decltype(unifex::await_transform), promise_type&, Value>)
-        -> callable_result_t<decltype(unifex::await_transform), promise_type&, Value> {
-      return unifex::await_transform(*this, (Value&&)value);
+    template <typename Value>
+    auto await_transform(Value&& value) -> decltype(auto) {
+      if constexpr (callable<decltype(unifex::await_transform), promise_type&, Value>) {
+        return unifex::await_transform(*this, (Value&&)value);
+      } else {
+        return Value((Value &&) value);
```

Comment:
Why are you forcing a copy of the argument here?
You should be able to return a `Value&&` safely.